tpl/plugins/init.go

```diff
@@ -0,0 +1,43 @@
+// Copyright 2018 The Hugo Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package plugins implements plugins of various sorts for Hugo templates.
+package plugins
+
+import (
+	"github.com/gohugoio/hugo/deps"
+	"github.com/gohugoio/hugo/tpl/internal"
+)
+
+const name = "plugins"
+
+func init() {
+	f := func(d *deps.Deps) *internal.TemplateFuncsNamespace {
+		ctx := New(d)
+
+		ns := &internal.TemplateFuncsNamespace{
+			Name:    name,
+			Context: func(args ...interface{}) interface{} { return ctx },
+		}
+
+		ns.AddMethodMapping(ctx.Starlight,
+			[]string{"starlight"},
```

Comment:
I'm unclear what the alias here is for, if it's not to make `plugins.starlight` work in a template...

Reply:
The alias is there to make `starlight` work in a template. As in `where` vs `collections.Where` etc.

Reply:
oh I see... hmm ok.  What's your opinion?  I think I like keeping plugins namespaced, since they have code outside of the template, being more visible and searchable is nice.

Reply:
btw, I'm assuming here that other plugin schemes will at some point exist, otherwise we could just call it "plugin" ... but I presume someone's going to want to do their own lua thing or whatever.